src/main/java/seedu/timetable/Timetable.java

```diff
@@ -113,12 +119,77 @@ private void addModuleToList(Module module) {
         }
     }
 
+    /**
+     * Deletes the module from the list of modules taken by the user.
+     *
+     * @param module Module to be Deleted
+     * @see Module
+     */
+    public static void deleteModuleFromList(Module module) throws UniModsException {
+        String moduleCode = module.getModuleCode();
+        boolean hasModule = false;
+        for (int i = 0; i < modules.size(); i++) {
+            if (modules.get(i).getModuleCode().equals(moduleCode)) {
+                hasModule = true;
+                modules.remove(modules.get(i));
+                TextUi.printModuleDeleted(moduleCode);
+                deleteFromLessons(module);
+            }
+        }
+        if (!hasModule) {
+            throw new UniModsException(TextUi.ERROR_MODULE_NOT_FOUND);
+        }
+    }
+
+   
+    public static void deleteFromLessons(Module module) {
+
+        for (int i = 0; i < 7; i++) {
+            String day = DAYS_OF_WEEK[i].toUpperCase();
+            switch (day) {
+            case "MONDAY":
+                deleteFromSchedule(module, monday);
+                break;
+            case "TUESDAY":
+                deleteFromSchedule(module, tuesday);
+                break;
+            case "WEDNESDAY":
+                deleteFromSchedule(module, wednesday);
+                break;
+            case "THURSDAY":
+                deleteFromSchedule(module, thursday);
+                break;
+            case "FRIDAY":
+                deleteFromSchedule(module, friday);
+                break;
+            case "SATURDAY":
+                deleteFromSchedule(module, saturday);
+                break;
+            case "SUNDAY":
+                deleteFromSchedule(module, sunday);
+                break;
+            default:
+                break;
+            }
+        }
```

Comment:
why not just call deleteFromSchedule on every schedule instead of doing a for loop